contrib/dracut/90zfs/mount-zfs.sh.in

```diff
@@ -58,7 +58,7 @@ ZFS_POOL="${ZFS_DATASET%%/*}"
 
 if import_pool "${ZFS_POOL}" ; then
 	# Load keys if we can or if we need to
-	if [ $(zpool list -H -o feature@encryption $(echo "${ZFS_POOL}" | awk -F\/ '{print $1}')) = 'active' ]; then
+	if [ "$(zpool list -H -o feature@encryption "$(echo "${ZFS_POOL}" | awk -F/ '{print $1}')")" = 'active' ]; then
```

Comment:
The double layers of quotes don't look right to me.  Are you sure this works as intended?

Reply:
I am. Those nested process substitutions are logically equivalent to
```sh
echo "${ZFS_POOL}" | awk -F/ '{print $1}'
zpool list -H -o feature@encryption "$()"
[ "$()" = 'active' ]
```
Each of which must be quoted separately, and each of which has its own execution/parsing context.

If `ZFS_POOL="pool name/dataset name"`, the previous code would evaulate, in turn, to
```sh
if [ $(zpool list -H -o feature@encryption $(echo "pool name/dataset name" | awk -F\/ '{print $1}')) = 'active' ]; then 
if [ $(zpool list -H -o feature@encryption pool name) = 'active' ]; then 
if [ = 'active' ]; then   # And "cannot open 'pool': no such pool", "cannot open 'name': no such pool" in stderr
if false; then   # And "sh: 1: [: missing ]" in stderr
```
whereas this code to
```sh
if [ "$(zpool list -H -o feature@encryption "$(echo "pool name/dataset name" | awk -F/ '{print $1}')")" = 'active' ]; then 
if [ "$(zpool list -H -o feature@encryption "pool name")" = 'active' ]; then 
if [ "active" = 'active' ]; then 
if true; then 
```